stat_summary_plots.py

```diff
@@ -0,0 +1,116 @@
+import sklearn as sk
+from sklearn.linear_model import LinearRegression
+import numpy as np
+from sklearn import datasets
+from numpy.linalg import det
+from numpy.linalg import inv
+from numpy import mat
+from numpy import random
+import matplotlib.pyplot as plt
+import seaborn as sns
+import pandas as pd
+import scipy.stats as ss
+
+
+class lr:
+    '''
+    implements the linear regression algorithm class
+    '''
+
+    def __init__(self, data_source, separator):
+        self.beta = np.matrix(np.zeros(2))
+        self.yhat = np.matrix(np.zeros(2))
+        self.r2 = 0.0
+        self.se = 0.0
+        self.f = 0.0
+        self.msr = 0.0
+        self.mse = 0.0
+        self.p = 0.0
+        data_mat = np.genfromtxt(data_source, delimiter=separator)
```

Comment:
I really like your idea and you did a great job implementing your solution. I  don't think you need to read the CSV file inside your class. You can assume that the user has already split the data into Features, Target(s) and you receive them as parameter. That's basically what @nirveshk module is doing and you can call his module and pass the results them to your class methods as parameters. This also helps your code become bug free due to some complications during reading data from a CSV file.

Reply:
Thanks! I'll look at it.